tests/circuit_drawer/test_draw_mpl.py

```diff
@@ -0,0 +1,523 @@
+        assert len(ax.patches) == 2
+        assert ax.patches[0].center == (layer, 0)
+        assert ax.patches[1].center == (layer, 1)
+
+        plt.close()
+
+controlled_data = [(qml.CY(wires=(0,1)), "Y"), 
+    (qml.CRX(1.2345, wires=(0,1)), "RX"),
+    (qml.CRot(1.2, 2.2, 3.3, wires=(0,1)), "Rot")
+]
+
+class TestControlledGates:
+    """Tests generic controlled gates"""
+
+    @pytest.mark.parametrize("op, label", controlled_data)
+    def test_control_gates(self, op, label):
+        """Test a variety of non-special gates.  Checks control wires are drawn, and
+        box drawn over target wires."""
+
+        with QuantumTape() as tape:
+            qml.apply(op)
+
+        _, ax = draw_mpl(tape)
+        layer = 0
+
+        assert isinstance(ax.patches[0], mpl.patches.Circle)
+        assert ax.patches[0].center == (layer,0)
+
+        control_line = ax.lines[2]
+        assert control_line.get_data() == ((layer,layer), (0,1))
+
+        assert isinstance(ax.patches[1], mpl.patches.Rectangle)
+        assert ax.patches[1].get_xy() == (layer-0.4, 0.6)
+
+        # two wire labels, so [2] is box gate label
+        assert ax.texts[2].get_text() == label
+
+        # box and text must be raised above control wire
+        # text raised over box
+        assert ax.patches[1].get_zorder() > control_line.get_zorder()
+        assert ax.texts[2].get_zorder() > ax.patches[1].get_zorder()
+
+        plt.close()
+
+    def test_CRX_decimals(self):
+        """Test a controlled parametric operation with specified decimals."""
+
+        with QuantumTape() as tape:
+            qml.CRX(1.234, wires=(0, 1))
+
+        _, ax = draw_mpl(tape, decimals=2)
+
+        # two wire labels, so CRX is third text object
+        assert ax.texts[2].get_text() == "RX\n(1.23)"
+        plt.close()
+
+general_op_data = [qml.RX(1.234, wires=0),
+    qml.Hadamard(0),
+    qml.S(wires=0),
+    qml.IsingXX(1.234, wires=(0,1)),
+    qml.U3(1.234,2.345,3.456, wires=0),
+    ### Templates
+    qml.QFT(wires=range(3)),
+    qml.Permute([4,2,0,1,3], wires=(0,1,2,3,4)),
+    qml.GroverOperator(wires=(0,1,2,3,4,5)),
+    ### Continuous Variable
+    qml.Kerr(1.234, wires=0),
+    qml.Beamsplitter(1.234,2.345, wires=(0,1)),
+    qml.Rotation(1.234, wires=0)
+]
+
+class TestGeneralOperations:
+    """Tests general operations."""
+
+    @pytest.mark.parametrize("op", general_op_data)
+    def test_general_operations(self, op):
+        """Test a variety of operations produce a rectangle accross relevant wires
+        and correct label text."""
+
+        with QuantumTape() as tape:
+            qml.apply(op)
+
+        _, ax = draw_mpl(tape)
+
+        num_wires = len(op.wires)
+        assert ax.texts[num_wires].get_text() == op.label()
+
+        assert isinstance(ax.patches[0], mpl.patches.Rectangle)
+        assert ax.patches[0].get_xy() == (-0.4, -0.4)
+        assert ax.patches[0].get_width() == 0.8
+        assert ax.patches[0].get_height() == num_wires-0.2
```

Comment:
Just to double-check: do all of the operations in `general_op_data` produce the same sized box?